mlir/lib/Conversion/TosaToMIOpen/TosaToMIOpen.cpp

```diff
@@ -297,148 +281,89 @@ class MatMulConverter final : public OpConversionPattern<tosa::MatMulOp> {
   }
 };
 
-class TransposeConverter final : public OpConversionPattern<tosa::TransposeOp> {
-public:
-  using OpConversionPattern<tosa::TransposeOp>::OpConversionPattern;
+struct TransposeRewritePattern : public OpRewritePattern<tosa::TransposeOp> {
+  using OpRewritePattern<tosa::TransposeOp>::OpRewritePattern;
 
-  bool transposeEqualTo(SmallVector<int64_t> &given,
-                        SmallVector<int64_t> &perm) const {
-    for (uint32_t i = 0; i < 4; i++) {
-      if (perm[i] != given[i])
-        return false;
+  SmallVector<int32_t> getTransposeDims(Value v) const {
+    if (Operation *cval = v.getDefiningOp<arith::ConstantOp>()) {
+      auto cattr = cval->getAttr("value").cast<DenseElementsAttr>();
+      return SmallVector<int32_t>(cattr.getValues<int64_t>());
+    }
+    if (Operation *cval = v.getDefiningOp<tosa::ConstOp>()) {
+      auto cattr = cval->getAttr("value").cast<DenseElementsAttr>();
+      return SmallVector<int32_t>(cattr.getValues<int32_t>());
     }
-    // all matches.
-    return true;
+    // May be bufferization cast
+    //  but this is no longer a bufferization pass, so assert
+    assert(0);
```

Comment:
Nit: can we get an error in the assert?

Reply:
should not encounter this case and the message is in the comment.